src/librustc_mir/build/matches/mod.rs

```diff
@@ -678,6 +715,23 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
 
         let arm_block = arm_blocks.blocks[candidate.arm_index];
 
+        // link previously generated false edge to the enter block
+        let dummy_source_info = self.source_info(DUMMY_SP);
+        if let Some(imaginary_target) = opt_imaginary_target_block.take() {
```

Comment:
~~I think it would be cleaner to use `mem::replace(opt_imaginary_target_block, Some(imaginary_target_block))` here